Feature summary

Eden-Alem opened the pull request:
Added summarize_text function in pike_tools/summaries.py
-         Uses Google Gemini 1.5 Flash model via google.generativeai
-         Renders prompts using a Jinja2 template (templates/summarize.j2)
-         Extracts structured JSON output and returns the last "Denser_Summary"
-         Implements error handling for invalid JSON responses

Added tests in tests/test_tool_summaries.py:
-     test_summarize_text_mocked: uses monkeypatching to simulate Gemini model and template environment
-     test_summarize_text_real: calls real model response, decorated with @pytest.mark.skip_in_pipeline for CI exclusion

Fixed ModuleNotFoundError: No module named 'tools'
-     Issue was caused by an import conflict between the fitz package and a local tools directory
-     Resolved by renaming the local tools module to pike_tools to avoid conflict with fitz's internal import
-     Added additional packages to requirements.txt (fitz, tools, jinja2)

Eden-Alem committed:
feat: improve summarize_text tool with robust JSON parsing and structured output handling

Eden-Alem committed:
feat: improve summarize_text tool with robust JSON parsing and structured output handling

Eden-Alem committed:
feat: improve summarize_text tool with robust JSON parsing and structured output handling

Eden-Alem committed:
feat: improve summarize_text tool with robust JSON parsing and structured output handling

Eden-Alem committed:
feat: improve summarize_text tool with robust JSON parsing and structured output handling

Eden-Alem committed:
feat: improve summarize_text tool with robust JSON parsing and structured output handling

Eden-Alem committed:
fix: handle empty and invalid JSON responses from Gemini API

Eden-Alem committed:
fix: handle empty and invalid JSON responses from Gemini API

Eden-Alem committed:
fix: handle empty and invalid JSON responses from Gemini API

Eden-Alem committed:
fix: handle empty and invalid JSON responses from Gemini API

Eden-Alem committed:
Add unit tests for summarize_text tool with mock and optional real API test. Update import paths and fixed naming conflict with 'tools' module

Eden-Alem committed:
Add unit tests for summarize_text tool with mock and optional real API test. Update import paths and fixed naming conflict with 'tools' module

Eden-Alem committed:
Add unit tests for summarize_text tool with mock and optional real API test. Update import paths and fixed naming conflict with 'tools' module

Eden-Alem committed:
Add unit tests for summarize_text tool with mock and optional real API test. Update import paths and fixed naming conflict with 'tools' module

Eden-Alem committed:
Add unit tests for summarize_text tool with mock and optional real API test. Update import paths and fixed naming conflict with 'tools' module

Eden-Alem committed:
Add unit tests for summarize_text tool with mock and optional real API test. Update import paths and fixed naming conflict with 'tools' module

renatkh committed:
chore: moved tests

feat: added setup.sh, start.sh, and docs/screenshot.png

chore: updated readme.md

feat: updated License to MIT.

Eden-Alem committed:
fix: install tools and rename folder back to tools

Eden-Alem committed:
fix: install tools and rename folder back to tools

Eden-Alem committed:
fix: install tools and rename folder back to tools

Eden-Alem committed:
fix: install tools and rename folder back to tools

Eden-Alem committed:
fix: install tools and rename folder back to tools

Eden-Alem committed:
fix: install tools and rename folder back to tools

Eden-Alem committed:
fix: remove redundant packages

Eden-Alem committed:
fix: remove package

Eden-Alem committed:
fix: remove package

Eden-Alem committed:
fix: remove package

Eden-Alem committed:
fix: remove package

ChiKonstantin committed:
feat: Fe be integration

* Added frontend to repo

* connected current FE features to BE, changed some mocked data and naming on the BE

* fix: Update .gitignore to quit pushing node_modules directory.

* fix:  Changed modify_chat_status to return the right information.  Updated unit tests to reflect changes to returned info.

* fix: renamed get_user_pinned_chats

* fix:  Remove debug code and fix api test.

* fix: backend comment removed, some names changed to camelCase

* fix:  Remove deletion flag keys for now.

* fix: cleaned up some commented out code and comments in general, lets merge it!

---------

Co-authored-by: Justin Hooper <jhooperonline@gmail.com>

renatkh committed:
feat: updated scripts to use in projevt nvm install

JHooperCode committed:
feat: Update setup script to verify correct version of python, install .nvm under frontend directory, not affect user's .bashrc, and generate a startup script with the appropriate NVM directory set as necessary.

renatkh committed:
feat: added gitignore

ChiKonstantin committed:
feat: Add CreateChatButton and PopupBlanket components; update Chat and ChatsWindow styles

ChiKonstantin committed:
feat: Implement debounced input handling in CreateChatButton; update styles for input field and button interactions

ChiKonstantin committed:
feat: Enhance chat functionality with new SendButton, update state management for new messages, and style improvements for input fields

ChiKonstantin committed:
feat: Implement delayed expansion for CreateChatButton and enhance PopupBlanket visibility; update styles for smoother transitions

ChiKonstantin committed:
feat: Enhance CreateChatButton and SendButton components; add icon, improve styles, and update layout for better user experience

ChiKonstantin committed:
feat: Update CreateChatButton styles; enhance button appearance and interactions for better user experience

ChiKonstantin committed:
feat: Enhance CreateChatButton functionality; add textarea focus on expansion and refactor code for better readability

ChiKonstantin committed:
feat: Update CreateChatButton styles; enhance background overlay and transition effects for improved user experience

ChiKonstantin committed:
feat: Refactor chat components; rename class for clarity, remove debug log, and enhance CreateChatButton styles for better layout and responsiveness

ChiKonstantin committed:
refactor: Update comments for clarity in CreateChatButton and remove unused styles in SendButton

ChiKonstantin committed:
feat: Add createNewChat thunk to handle chat creation with UUID generation

renatkh committed:
feat: added client generate chatId to create_chat; return chat_interface and response message; updated tests

chore: renamed all *_id to *Id for consistency

ChiKonstantin committed:
feat: implement smooth scrolling to chat area and enhance chat creation with retry logic

ChiKonstantin committed:
feat: implement chat history fetching and display in Chat component; update store to manage chat history

renatkh committed:
chore: formatting

feat: switched new_chat response to dict

renatkh committed:
feat: changed mocke_massage_interface

ChiKonstantin committed:
feat: refactor createNewChat to use message object instead of newChat

ChiKonstantin committed:
feat: update chat history fetching to handle message structure and add logging

JHooperCode committed:
feat:  Add basic webpage text extraction.

ChiKonstantin committed:
fix: reduce delay for textarea adjustment and update chat button styles

JHooperCode committed:
feature: Implement Chat base classes and fe serialization.

ChiKonstantin committed:
feat: implement UserMessage and AgentMessage components; update Chat to render messages based on type and adjust styles

ChiKonstantin committed:
style: adjust formatting of chat area div for improved readability

renatkh commented on backend/src/templates/summarize.j2:
```diff
@@ -0,0 +1,29 @@
+You will generate increasingly concise, entity-dense summaries of the following article.
+
+Repeat the following two steps five times:
```
add number of steps as a parameter. I remember that they have 5 in the paper, but turns out that the best performance is achieved with 2 steps

Eden-Alem replied:
I can look into the summaries but there were changes step by step I only use the last summary at the end

Eden-Alem replied:
{'Missing_Entities': 'Global industries; Automation', 'Denser_Summary': "Artificial intelligence is impacting various aspects of global industries.  It's doing this through automation and other means, leading to significant changes.  The effects are widespread and far-reaching, affecting numerous sectors.  Further research is needed to fully understand the implications.  This transformative technology is reshaping our world."}
{'Missing_Entities': 'Data analysis', 'Denser_Summary': "AI's impact on global industries is driven by automation and data analysis.  These changes are significant and far-reaching, affecting numerous sectors.  The technology is reshaping our world through improved decision-making support.  Further research is needed to fully understand the implications."}
{'Missing_Entities': 'Decision-making support', 'Denser_Summary': 'AI reshapes global industries via automation, data analysis, and improved decision-making.  Its impact is significant and widespread, affecting numerous sectors. This transformative technology is reshaping our world.'}
{'Missing_Entities': None, 'Denser_Summary': 'AI transforms global industries through automation, data analysis, and decision-making support. Its impact is significant and widespread, affecting numerous sectors.'}
{'Missing_Entities': None, 'Denser_Summary': 'AI reshapes global industries: automation, data analysis, decision support.'}


------
The above is an example output I think 5 is ideal. Don't you think?

renatkh replied:
it could be, but we should still make it a parameter that we can tune in the end.

renatkh replied:
if the output is a single response, how did you generate the above? did you loop through it?

renatkh commented on backend/src/templates/summarize.j2:
call it CoD_summarize

renatkh replied:
it is possible, but in this case I think it would be a bit more specific. Who knows, maybe in a year there will be a new summary technique. Also, keep in mind, that this summary technique is slow and expensive. In simple cases, <500 words we may want to invoke a simple summary.

renatkh commented on backend/src/tools/summaries.py:
```diff
@@ -1,7 +1,48 @@
-    return "This is a summary of the text (mocked)."
+def summarize_text(input_text: str) -> str:
+    """
+    Summarize a given text using a language model and a Jinja2 prompt template.
```
no need to say language model in jinja prompt. This docstring is used by LLM, so it needs only info necessary to understand what it does, but not the implementation details.

renatkh commented on backend/src/tools/summaries.py:
```diff
@@ -1,7 +1,48 @@
+    """
+    Summarize a given text using a language model and a Jinja2 prompt template.
+
+    This function leverages Google's Gemini model to generate a dense summary
```
ditto

renatkh commented on backend/src/tools/summaries.py:
```diff
@@ -1,7 +1,48 @@
+    str
+        A concise summary of the input text, returned from the language model.
+    """
+    genai.configure(api_key=os.getenv("GOOGLE_API_KEY"))
```
you are not using langchain here?

renatkh commented on backend/src/tools/summaries.py:
```diff
@@ -1,7 +1,48 @@
 import langchain_core.tools as lcct
+import google.generativeai as genai
```
why are you not using langchain google implementation? Take a look at my `test_graph_builder.test_gemini_model_calls_tool` for an example

renatkh commented on backend/src/tools/summaries.py:
```diff
@@ -1,7 +1,48 @@
+    template = env.get_template("templates/summarize.j2")
+
+    prompt = template.render(input_text=input_text)
+    model = genai.GenerativeModel("gemini-1.5-flash")
```
make sure to use langchain where appropriate.

renatkh commented on tests/test_tool_summaries.py:
```diff
@@ -0,0 +1,61 @@
+    output = su.summarize_text(input_text)
+
+    assert isinstance(output, str)
+    print("\nReal summary output:", output)
```
tests should never print

renatkh commented on tests/test_tool_summaries.py:
```diff
@@ -0,0 +1,61 @@
+
+    return captured
+
+def test_summarize_text_mocked(mock_dependencies):
```
not sure what are you testing here, we may need to discuss it.

renatkh commented on tests/test_tool_summaries.py:
```diff
@@ -0,0 +1,61 @@
+    input_text = "Artificial Intelligence is changing global industries with automation, data analysis, and decision-making support."
+    output = su.summarize_text(input_text)
+
+    assert isinstance(output, str)
```
we may want to assert for something specific. I think the prompt returns multiple summaries that need to be dropped. Did you check?

Eden-Alem replied:
Yes I was printing and looking at the output its a single response.

JHooperCode committed:
chore:  Remove unused code, rename variables to be clearer, clean up tests and add url sanitization/related test.

JHooperCode committed:
fix:  Add in a direct host resolution check and fix url validation to not spew pages of message errors for ill-formed URLs

ChiKonstantin committed:
feat: add CollapseChatsButton component and functionality to collapse all chats; update styles and store actions

JHooperCode committed:
chore:  Revert descendant check for tags in webpage parsing back to use the more expansive isinstance formulation.  Fix  local file test fixture to circumvent the host_resolution check

ChiKonstantin committed:
feat: add ChatMessageInput component for sending messages; update Chat and styles for improved message input handling

ChiKonstantin committed:
refactor: update Chat, ChatMessageInput, and CreateChatButton components for improved structure and styling; enhance logging and adjust styles for better usability

ChiKonstantin committed:
refactor: improve chat component logic and optimize chat history fetching; update styles for better layout and usability

ChiKonstantin committed:
feat: implement message appending functionality in chat; update Chat and ChatMessageInput components for improved message handling

ChiKonstantin committed:
feat: implement sendChatMessage thunk for sending messages; update Chat and ChatMessageInput components for improved message handling and scrolling

ChiKonstantin committed:
refactor: remove commented-out code and update styles for improved consistency in ChatsWindow and Dashboard components

JHooperCode committed:
chore: Update docstring for host connection exception thrown.

JHooperCode committed:
fix:  Update the messages type in ChatHistory to leverage langgraph's automatic message history.

ChiKonstantin committed:
refactor: simplify Chat component structure and update styles for improved layout and usability

ChiKonstantin committed:
refactor: reorder props in Chat component for improved readability; remove commented console log in CreateChatButton

JHooperCode committed:
fix:  Remove unused embedding object from webpage parser.

ChiKonstantin committed:
refactor: remove commented code and improve readability in ChatMessageInput, CreateChatButton, Dashboard, UserLogin, and store files

ChiKonstantin committed:
refactor: enhance structure and styling of agent buttons for improved layout and consistency

ChiKonstantin committed:
refactor: add agent icon to AgentButton and update styles for improved hover effects

JHooperCode committed:
feature:  Implement pydantic based Chat object toward building out from mock to real architecture.

ChiKonstantin committed:
refactor: update chat collapsing logic to allow collapsing all chats or only the last chat

ChiKonstantin committed:
refactor: update chat IDs in mock chat interfaces for consistency

JHooperCode committed:
chore:  Debugging the routes using chat objects.  (Partially done.)

ChiKonstantin committed:
fix: smooth scroll on chat

renatkh committed:
feat: turned state into chat

feat: removed StateFull

feat: updated chet_history, create_chat, and get_responce routes to use chats

feat: use create_react_agent instead of graph builder.

chore: fix some tests

renatkh committed:
fix: create_chat return message

renatkh committed:
feat: returned use of graph_builder

fix: fixed create_chat test

Eden-Alem committed:
feat: add a pydantic model interface object

JHooperCode committed:
fix:  Fix pytest fixtures and tests: get_chat_history, modify_chat_status

JHooperCode committed:
fix:  Fix test_graph_builder test by modifying build_graph call to directly include tool list and graph_builder.tool_condition so that it returns correctly when tool_calls are present.

JHooperCode committed:
fix:  Update test_gemini_model_calls_tool to work iwth the updated version of build_graph where tool lists are passed directly to the build command.  Cleaned up unnecessary monkeypatch inclusions in graph_builder.

JHooperCode committed:
fix:  Update mock model to account for more complex langchain messages being passed in.

JHooperCode committed:
fix:  Get response test now working (with Gemini 2.0.

Eden-Alem committed:
fix: move test file location, use langchain google gen ai, update docstring and update jinja template file name to include promting technique used

Eden-Alem committed:
fix: move test file location, use langchain google gen ai, update docstring and update jinja template file name to include promting technique used

Eden-Alem committed:
fix: move test file location, use langchain google gen ai, update docstring and update jinja template file name to include promting technique used

Eden-Alem committed:
fix: move test file location, use langchain google gen ai, update docstring and update jinja template file name to include promting technique used

Eden-Alem committed:
fix: move test file location, use langchain google gen ai, update docstring and update jinja template file name to include promting technique used

Eden-Alem committed:
fix: move test file location, use langchain google gen ai, update docstring and update jinja template file name to include promting technique used

renatkh committed:
feat: CHAT_CACH is autopopulation on chat creation

fix: get_response test uses mock LLM

renatkh committed:
fix: fixture was missing import

fix: import in test_chat

JHooperCode committed:
fix:  Bug touchup in mocks that was breaking backend.

renatkh committed:
feat: added default_oai agent

renatkh committed:
feat: added Agent and AgentConfig objects

feat: added create_agent route and tests

feat: updated create_chat to use default agent

Eden-Alem committed:
fix: remove defaults

renatkh committed:
fix: addressed comments

renatkh committed:
fix: revert __context removal

renatkh committed:
feat: added .env template

renatnova committed:
feat: integrated model object; restructure

JHooperCode committed:
chore: Update docstring for maximal clarity.

ChiKonstantin committed:
chore: update screenshot for documentation

ChiKonstantin committed:
fix: adjust message bubble margins for better alignment

renatnova committed:
feat: added message formatting including markdown

ChiKonstantin committed:
fix: update package name and add react-markdown dependency

ChiKonstantin committed:
refactor: improve message handling in AgentMessage and integrate MarkdownDemo in Chat

ChiKonstantin committed:
refactor: remove MarkdownDemo component and clean up AgentButton and Chat components

ChiKonstantin committed:
fix: enhance agent button interactivity with cursor and user-select styles

ChiKonstantin committed:
fix: add z-index to collapse chats button for improved stacking context

ChiKonstantin committed:
fix: update font-family for create chat button to improve consistency across platforms

JHooperCode committed:
Chore:  Move icons to tool_class branch.

JHooperCode committed:
refactor:  Changed model.get_default_model to actually return the model labelled "default".  If no such model exists, model.create_default_model is called first to build a default, and this default is registered and returned.

Eden-Alem committed:
fix: add iteration number as an input variable

Eden-Alem committed:
fix: add iteration number as an input variable

Eden-Alem committed:
fix: add iteration number as an input variable

Eden-Alem committed:
fix: add iteration number as an input variable

Eden-Alem committed:
fix: add iteration number as an input variable

renatkh reviewed (approved):
thank you!

renatkh committed:
feat: added dark them

renatkh committed:
bug: fixed button selection

JHooperCode committed:
feature:  Implementation of pike_tool as a drop in tool replacement to carry info for PIKE as well as the base tool info for langchain.

JHooperCode committed:
fix:  Updated the pike_tool decorator to at least work with no arguments in the decorator declaration.  Not fully implemented for the complex cases covered by langchain_core.tools @tool decorator, however.

JHooperCode committed:
chore:  Reverted non-theme related changes from style.css.

JHooperCode committed:
fix:  Changed collapse all chats functions to actually collapse all chats.  Functionality for 'most recent' is more complicated than what was implemented.

JHooperCode committed:
fix:  Make @pike_tool call always require arguments to differentiate from expectations of full langchain @tool compatibility with differing call signatures.

JHooperCode committed:
fix:  Add display member to PikeTool class.

JHooperCode committed:
Chore:  Minor clean-up pre merge.

JHooperCode committed:
Fix:  Tweak requirements.txt to fix issue.

JHooperCode committed:
Fix:  Tweak requirements.txt to fix issue.

JHooperCode committed:
Merge branch 'feature-summary' of https://github.com/Sentelex/pike into feature-summary

JHooperCode committed:
Fix:  Clean up summaries.py to use a pike model internally.  Clean up tests to account for this more complicated structure.

JHooperCode committed:
Fix:  Clean up summaries.py to use a pike model internally.  Clean up tests to account for this more complicated structure.

JHooperCode committed:
Fix:  Clean up summaries.py to use a pike model internally.  Clean up tests to account for this more complicated structure.

JHooperCode committed:
Fix:  Fixes tool path issues and adds explicit arg_schema for Google tool requirements.

JHooperCode committed:
Fix:  Undo change attempting to swap out ID mapping file retrieval for direct file passing.

JHooperCode committed:
fix:  Remove code that had model instantiations set the model environment variable keys to ensure they were available.  This is not a good model and was causing edge-case testing failures.

JHooperCode committed:
fix:  Remove code that had model instantiations set the model environment variable keys to ensure they were available.  This is not a good model and was causing edge-case testing failures.

JHooperCode committed:
fix:  Removed API key storage from models/model environment to improve security.  Fixed multiple path errors.

JHooperCode committed:
fix:  Removed API key storage from models/model environment to improve security.  Fixed multiple path errors.

JHooperCode committed:
fix:  Removed API key storage from models/model environment to improve security.  Fixed multiple path errors.

JHooperCode committed:
Fix:  Update messed up rebase commit.

JHooperCode committed:
Merge branch 'feature-summary' of https://github.com/Sentelex/pike into feature-summary

JHooperCode committed:
Merge branch 'feature-summary' of https://github.com/Sentelex/pike into feature-summary

JHooperCode committed:
fix:  Refactored environment variable extraction into pike_util.py.  Cleaned up model.py and summaries to not leave environment keys all around.  Added arg_schema to summaries.py for Google compatibility.
